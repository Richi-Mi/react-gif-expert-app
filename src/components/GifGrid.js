import React, { Fragment } from 'react'
import PropTypes from 'prop-types';
import useFetchGifs from '../hooks/useFetchGifs'
import GifGridItem from './GifGridItem'

const GifGrid = ({ category }) => {
    // const [images, setImages] = useState([])
    // useEffect(() => {
    //     getGifs( category )
    //         .then( imgs => setImages( imgs ))
    // }, [ category ])

    // useEffect( funcion(), dependencias ) .- Va a ejecutar el codigo, solo si se cambia de estado
    // una dependencia, en este caso lo ponemos para que solo se ejecute una vez

    // Pusimos useEffect, en el hook, useFetchGifs

    const { data: images, loading } = useFetchGifs( category )

    return (
        <Fragment>
            <h3 className='animate__animated animate__fadeIn'> {category} </h3>
            { loading && <p className='animate__animated animate__flash'> Cargando... </p> }
            <div className='card-grid'>

                {
                    images.map(img => {
                        // { ...img } .- Mandamos las propiedades del objeto
                        return <GifGridItem key={img.id} {...img} />
                    })
                }

            </div>
        </Fragment>
    )
}
GifGrid.propTypes = {
    category: PropTypes.string.isRequired
}
export default GifGrid