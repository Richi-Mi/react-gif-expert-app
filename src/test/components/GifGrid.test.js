import '@testing-library/jest-dom';

import React from 'react';
import GifGrid from '../../components/GifGrid';

import { shallow } from 'enzyme';
import useFetchGifs from '../../hooks/useFetchGifs';

jest.mock('../../hooks/useFetchGifs');

describe(' 4.- Pruebas en el <GifGridItem /> ', () => {
    const category = 'One Punch Man';
    
    test(' * Debe de Mostrar el componente correctamente', () => {
        useFetchGifs.mockReturnValue({ // Configuramos cual va a ser la información retornada
            data: [],
            loading: true
        });

        const wrapper = shallow( <GifGrid category={ category } /> );

        expect( wrapper ).toMatchSnapshot();
    })
    test(' * Debe de Mostrar imagenes cuando se carge useFetchGifs()', () => {
        const gifs = [
            {
                id: '2022-01-10-2:03',
                title: 'One Punch Man',
                url: 'https://giphy.com/gif/aw0osae0ad21.gif'
            },
            {
                id: '2022-01-10-2:04',
                title: 'One Punch Man Two',
                url: 'https://giphy.com/gif/aw0osae0ad21.gif'
            }
        ]
        useFetchGifs.mockReturnValue({ // Configuramos cual va a ser la información retornada
            data: gifs,
            loading: false
        });
        const wrapper = shallow( <GifGrid category={ category } /> );

        expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find('p').exists() ).toBe( false );
        expect( wrapper.find('GifGridItem').length ).toBe( gifs.length );
    })
    
})
