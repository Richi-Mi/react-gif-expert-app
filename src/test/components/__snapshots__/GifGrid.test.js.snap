// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 4.- Pruebas en el <GifGridItem />   * Debe de Mostrar el componente correctamente 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
     
    One Punch Man
     
  </h3>
  <p
    className="animate__animated animate__flash"
  >
     Cargando... 
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;

exports[` 4.- Pruebas en el <GifGridItem />   * Debe de Mostrar imagenes cuando se carge useFetchGifs() 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
     
    One Punch Man
     
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="2022-01-10-2:03"
      key="2022-01-10-2:03"
      title="One Punch Man"
      url="https://giphy.com/gif/aw0osae0ad21.gif"
    />
    <GifGridItem
      id="2022-01-10-2:04"
      key="2022-01-10-2:04"
      title="One Punch Man Two"
      url="https://giphy.com/gif/aw0osae0ad21.gif"
    />
  </div>
</Fragment>
`;
